src/components/base-control.js

```diff
@@ -58,11 +58,21 @@ export default class BaseControl extends Component {
     super(props);
 
     this._events = null;
+    this._containerRef = null;
 
     this._onContainerLoad = this._onContainerLoad.bind(this);
   }
 
+  componentWillUnmount() {
+    const {eventManager} = this.context;
+    if (eventManager && this._events) {
+      eventManager.off(this._events);
+    }
+  }
```

Comment:
@Pessimistress Seemed like it made sense to remove events if the control is unmounted – let me know if I'm missing something.